src/shared/mount-util.c

```diff
@@ -772,6 +774,27 @@ int bind_mount_in_namespace(
         assert(src);
         assert(dest);
 
+        r = namespace_open(target, NULL, &mntns_fd, NULL, NULL, &root_fd);
+        if (r < 0)
+                return r;
```

Comment:
you are not initializing `*error_message*` here. on purpose or accidental omission?

btw, I think the whole `*error_message` stuff is something I'd drop. sure, the refactor you did here would generate slightly worse error messages towards the client, but so what... I'd just make sure that this function generates recognizable error codes for the *expected* failures (i.e. if the destination mount point doesn't exist, or can't be created or so), but for the others it's ok to just propagate the error code as is. Then, the stuff you currently write to `*error_message` just output via log_debug_errno() and we should be good.

This means: "expected" errors would still be handled nicely, and "unexpected" errors would still be debuggable easily once you turn on debug logging.

Reply:
Ok, refactored all of this to just log_debug_errno instead.